assets/javascripts/modules/analytics/omniture.js

```diff
@@ -18,73 +17,72 @@ define([
 
     function trackLink(link) {
         omniture = omniture || init();
-        omniture.then(function(){
+        omniture.then(function () {
             s.tl(true, 'o', link);
         });
     }
 
-    function trackEvent(prop17, pageName, products) {
-        omniture = omniture || init();
-
-        omniture.then(function(){
-            if (prop17) {
-                s.prop17 = prop17;
-                s.pageName = pageName;
-                s.products = products;
-                s.t();
-            }
-        });
-    }
-
-    function onSuccess() {
-        var s_code;
-
-        window.s_account = 'guardiangu-subscribe,guardiangu-network';
-        s = s_gi('guardiangu-network');
+    function setTracker(s) {
+        var pageInfo = guardian.pageInfo,
+            productData = guardian.productData;
 
-        s.pageName = document.title;
-        s.channel = 'Subscriber';
-        s.prop3 = 'GU.co.uk';
-        s.prop5 = 'Subscriber';
+        s.pageName = pageInfo.name;
+        s.channel = pageInfo.channel;
+        s.prop3 = pageInfo.publication;
+        s.prop5 = pageInfo.commissioningDesk;
         s.prop11 = 'Subscriber';
-        s.hier2 = 'GU/News/Subscriber/';
         s.prop14 = '24.0';
-        s.prop19 = 'Subscriber';
+        s.prop19 = pageInfo.product;
         s.eVar19 = 'D=c19';
         s.prop42 = 'Subscriber';
-        s.prop47 = 'UK';
+        s.prop47 = pageInfo.edition;
 
-        var main = document.querySelector('main');
-        if(main) {
-            var attributes = [].slice.call(main.attributes);
-            attributes.forEach(function(attr){
-                if(attr.name.indexOf('data-tracking-')>-1){
-                    var prop = attr.name.replace('data-tracking-','');
-                    s[prop] = attr.nodeValue;
-                }
-            });
+        if (productData) {
+            s.products = [
+                productData.source,
+                productData.type,
+                productData.frequency,
+                productData.amount,
+                productData.eventName
+            ].join(';');
         }
+        if (guardian.slug) {
+            s.prop17 = guardian.slug;
+        }
+    }
 
-        s_code = s.t();
+    function triggerPageLoadEvent() {
+        omniture = omniture || init();
+        omniture.then(function () {
+            setTracker(s);
+            var s_code = s.t();
+            setTracker(s);
+            if (s_code) {
+                /*jslint evil: true */
```

Comment:
Not your code but `/*jslint evil: true */` can go as we don't use `jslint` anymore so this line does nothing.

*However* we should have a quick look (as another PR) into whether we can avoid using `document.write` completely.